src/ai/defaultai.cc

```diff
@@ -705,7 +705,9 @@ void DefaultAI::late_initialization() {
 			}
 
 			for (const auto& temp_position : prod.working_positions()) {
-				bo.positions.push_back(temp_position.first);
+				for (uint8_t i = 0; i < temp_position.second; i++) {
```

Comment:
Oh, I am confused by this insertion, can you please explain in plain words - I admit I am not fully familiar with this worker stuff.

Reply:
that was a nasty one, which led to a crash for the barbarian big inn for example. 
if we have a building with two workers of the same type (eg. lua declaration like "barbarians_inn_keeper=2" the old code just had only 1 position in BuildingObserver while the site observer had two which led to vector length mismatch error. 
This code now creates as much positions in the building observer as specified with the amount (temp_position.second)

Reply:
Oh, ok